mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,462 @@
+                from mmdet.apis import single_gpu_test
+                results = single_gpu_test(
+                    runner.model, self.val_dataloader, show=False)
+            elif isinstance(self.eval_hook, DistEvalHook):
+                from mmdet.apis import multi_gpu_test
+                results = multi_gpu_test(
+                    runner.model, self.val_dataloader, gpu_collect=True)
+
+        eval_results = self.val_dataset.evaluate(results, logger='silent')
+        metadata = dict(epoch=runner.epoch + 1, **eval_results)
+        return metadata
+
+    def _is_best_ckpt(self, metadata):
+        """Check if the current checkpoint has the best metric score.
+
+        Args:
+            metadata (dict): Metadata associated with the checkpoint.
+
+        Returns:
+            bool: Returns True, if the checkpoint has the best metric score.
+        """
+        keys = list(metadata.keys())
+        map_metrics = [key for key in keys if 'mAP' in key]
+        ar_metrics = [key for key in keys if 'AR' in key]
+        if len(map_metrics) > 0:
+            map_score = metadata.get(map_metrics[0], None)
+            return self._is_best_score(map_score)
+        elif len(ar_metrics) > 0:
+            ar_score = metadata.get(ar_metrics[0], None)
+            return self._is_best_score(ar_score)
+        else:
+            return False
+
+    def _is_best_score(self, score):
+        if score is None:
+            return
+
+        if score > self.best_score:
+            self.best_score = score
+            return True
+        else:
+            return False
+
+    def _init_data_table(self):
+        """Initialize the W&B Tables for validation data."""
+        columns = ['image_name', 'image']
+        self.data_table = self.wandb.Table(columns=columns)
+
+    def _init_pred_table(self):
+        """Initialize the W&B Tables for model evaluation."""
+        columns = ['epoch', 'image_name', 'ground_truth', 'prediction'] + list(
+            self.class_id_to_label.values())
+        self.eval_table = self.wandb.Table(columns=columns)
+
+    def _add_ground_truth(self):
+        num_total_images = len(self.val_dataset)
+        if self.num_eval_images > num_total_images:
+            warnings.warn(
+                'The num_eval_images is greater than the total number '
+                'of validation samples. The complete validation set '
+                'will be logged.', UserWarning)
+        self.num_eval_images = min(self.num_eval_images, num_total_images)
+
+        classes = self.val_dataset.get_classes()
+        self.class_id_to_label = {id: name for id, name in enumerate(classes)}
+        self.class_set = self.wandb.Classes([{
+            'id': id,
+            'name': name
+        } for id, name in self.class_id_to_label.items()])
+
+        from mmdet.datasets.pipelines import LoadImageFromFile
+        img_loader = LoadImageFromFile()
+        img_prefix = self.val_dataset.img_prefix
```

Comment:
We can directly use the `LoadImageFromFile` pipeline in dataset pipelines. It may have some special settings for different datasets.